engineering_tools/vector_map_scripts/osm_transform.py

```diff
@@ -1,3 +1,51 @@
+"""
+This script relocates and optionally rotates an OpenStreetMap (OSM) file that uses a custom geoReference. It's useful when you want to:
+- Move a local map to a new geographic location.
+- Preserve geometry, scale, and layout.
+- Rotate the map around its origin (for alignment or testing).
+---
+
+Steps:
+
+1. Read the input OSM XML file.
+2. Extract the original geoReference (a Transverse Mercator projection centered at some latitude/longitude).
+3. Convert each node’s lat/lon to projected (X, Y) coordinates using the original projection.
+4. Apply a 2D rotation (optional) around the local origin (0, 0).
+5. Transform the rotated (X, Y) into new lat/lon coordinates using a new projection centered at a new location.
+6. Update the <geoReference> tag to reflect the new center.
+7. Save the updated OSM XML file with transformed coordinates.
+
+---
+
+Inputs:
+
+1- input_file.osm: An OSM XML file that:
+- Contains a <geoReference> string using +proj=tmerc
+- Has <node> elements with lat and lon attributes
+
+2- output_file.osm: The transformed OSM map name that:
+- All node positions have been relocated and optionally rotated
+- The <geoReference> tag is updated to match the new map center
+- The map geometry is preserved in relative terms but relocated globally
+
+---
+
+Parameters to adjust to transform:
+
+- rotation_deg: Rotation angle in degrees (positive = counter-clockwise)
+- new_lat_0, new_lon_0: New center location in geographic coordinates (latitude, longitude)
+
+--
+
+Dependency:
+pip install lxml pyproj
+
+--
+how to run the script:
+
+python osm_transform.py suntrax.osm suntrax_transformed.osm
```

Comment:
Nit: Consider replacing `suntrax.osm` with `<input-filename>.osm` and `suntrax_transformed.osm` to `<output-filename>.osm`, or similar